WalletWasabi/Blockchain/TransactionBuilding/BranchAndBound.cs

```diff
@@ -121,7 +121,7 @@ private bool TryFindSolution(SelectionStrategy searchStrategy, [NotNullWhen(true
 
 	private NextAction GetRandomNextAction()
 	{
-		return _random.Next(0, 2) == 1 ? NextAction.IncludeFirstThenOmit : NextAction.OmitFirstThenInclude;
+		return _random.Next(1, 500) < 250 ? NextAction.IncludeFirstThenOmit : NextAction.OmitFirstThenInclude;
```

Comment:
Do we gain something with this? I'm not against this, just asking.

Reply:
Well, in theory it shouldn't. In practice when I debug tests it seems to matter a lot on my computer. Given that the random number generator here is just a pseudo one and not statistically correct one, it seems it matters.

Reply:
Nitpick. You can do 

```
public static bool NextBoolean(this Random random)
{
    return random.Next() > (Int32.MaxValue / 2);
    // Next() returns an int in the range [0..Int32.MaxValue]
}
```

Took it from here: https://stackoverflow.com/questions/19191058/fastest-way-to-generate-a-random-boolean

Reply:
From the article, it is saying: rand.NextDouble() is the fastest. 

`rand.NextDouble() >= 0.5`

Reply:
I see. Applied.